command/operator_diagnose.go

```diff
@@ -217,12 +219,29 @@ func (c *OperatorDiagnoseCommand) RunWithParsedFlags() int {
 	// TODO: check for storage backend
 	c.UI.Output(same_line + status_ok + phase)
 
+	var storageErrors []string
+
 	phase = "Access storage"
 	c.UI.Output(status_unknown + phase)
 	_, err = server.setupStorage(config)
+	storageErrors = append(storageErrors, err.Error())
 	if err != nil {
-		c.UI.Output(same_line + status_failed + phase)
-		c.UI.Output(err.Error())
+
+	}
+
+	// Initialize the Service Discovery, if there is one
```

Comment:
If storage setup above fails with an error, does it make sense proceed to the tls portion?

Reply:
It does, good call!